test_conformance/buffers/test_buffer_write.cpp

```diff
@@ -851,7 +883,9 @@ int test_buffer_write_struct( cl_device_id deviceID, cl_context context, cl_comm
 
                 if (gTestMap) {
                     void *dataPtr;
-                    dataPtr = clEnqueueMapBuffer(queue, buffers[ii], CL_TRUE, CL_MAP_WRITE, 0, ptrSizes[i]*num_elements, 0, NULL, NULL, &err);
+                    dataPtr = clEnqueueMapBuffer(
```

Comment:
Should the unmap on line 898 be updated to `buffers[0]` from `buffers[ii]` too?

Likewise the calls to `clSetKernelArg` use `buffers[ii]` on line 918+919(github won't let me comment directly on the lines)